src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepository.java

```diff
@@ -48,16 +50,25 @@
   public static final Pattern ADDITIONAL_PARAMETER_REGEX_PATTERN =
       Pattern.compile(ADDITIONAL_PARAMETER_REGEX);
 
-  private StashPullRequestsBuilder builder;
+  private AbstractProject<?, ?> job;
   private StashBuildTrigger trigger;
   private StashApiClient client;
 
-  public StashRepository(StashPullRequestsBuilder builder) {
-    this.builder = builder;
+  public StashRepository(@Nonnull AbstractProject<?, ?> job, @Nonnull StashBuildTrigger trigger) {
+    this(job, trigger, null);
+  }
+
+  // For unit tests only
+  public StashRepository(
```

Comment:
make it package private?

Reply:
actually maybe it would be better to just override init method in tests?

Reply:
I made it package private. I consider annotating it with VisibleForTesting, but that would require a dependency on Google Guava for very little benefit.

I believe overriding `init()` would require making a subclass of StashRepository, which I would rather avoid.

I'm actually going to remove `init()` when adding pipeline support. I think the best testable solution would be to use a class factory, but I'm trying to minimize changes in this PR.